### ISSUE TITLE
ChildController's in viewpager, onDetach, onDestroy called 

### ISSUE THREAD

**Issue by tonyjs** [bluelinelabs/Conductor#518]
first of all here is router pager adapter.
```
class PageAdatper : RouterPagerAdapter {
    override configureRouter(router: Router, position: Int) {
        if (!router.hasRootController()) {
            when (position) {
                0 -> router.setRoot(RouterTransaction.with(AController())
                1 -> router.setRoot(RouterTransaction.with(BController()))
                2 -> router.setRoot(RouterTransaction.with(CController()))
        }
```

now open `TestController` inside on `AController`
```
class AController : Controller {
    ...
    override onViewBound(view: View) {
        view.btn_test.onClick {
            getChildRouter(view.test_view_group).apply { setPopsLastView(true) }
                .pushController(RouterTransaction.with(TestController()))
        }
    }
    ...
}
```

then I pressed the back button on `TestController`,
i can see the log onDetach and onDestroy called in `BController` and `CController`

is this a bug? or intended?

if intended, which is recommended way to open inside controller on parent controller which insside viewpager?

**Comment by dimsuz** [comment]
I believe this happens due to the same reasons described in #314 

Namely: Conductor has a bug where it detaches child controllers even if parent controller has RETAIN_DETACH set - and never attaches them back afterwards.

@PaulWoitaschek  @EricKuck should this be closed as a duplicate?

**Comment by tonyjs** [comment]
yeah, similar. 
I've found the solution(confused, It is right way), 
do not push controller with child router inside viewpager(or already exist on child router), use parent router(maybe root router). and handle the backstack.
E.g
```
override fun handleBack() {
    return pageAdapter.getRouter(pager.currentItem).handleBack()
}
```

if making child router, must to manage the backstack. maybe, attach backstack into parent router's backstack.

**Comment by EricKuck** [comment]
Fixed via #571
